File: back-node/src/models/helper/user.helper.ts
import express from 'express';
import { ListFormat } from 'typescript';
import List from '../../utils/List';
import UserDTO from '../DTO/UserDTO';
import User from '../entity/User';


export default class UserHelper
{
    private UserHelper() {}

    public static convertAll(userList: List<User>): List<UserDTO>
    {
        let userDTOList: List<UserDTO> = new List<UserDTO>();

        userList.forEach((user:User) => {
            userDTOList.add(UserHelper.convert(user));
        })

        return userDTOList;
    }

    public static convert(user:User): UserDTO
    {
        return new UserDTO(
            user.getId(),
            user.getName(),
            user.getFirstname(),
            user.getPseudo(),
            user.getEmail(),
            user.getPassword(),
            user.getBirthdate(),
            user.getBase64(),
            user.getStatus()
        );
    }
}

File: back/src/main/java/com/project/booktime/model/helper/UserHelper.java
package com.project.booktime.model.helper;

import com.project.booktime.model.dto.UserDTO;
import com.project.booktime.model.entity.User;

import java.util.ArrayList;
import java.util.List;

public class UserHelper {

    private UserHelper() { }

    public static List<UserDTO> convertAll(List<User> userList) {
        List<UserDTO> userDTOList = new ArrayList<>();

        for (User user : userList) {
            userDTOList.add(convert(user));
        }

        return userDTOList;
    }

    public static UserDTO convert(User user) {
        return new UserDTO(
                user.getId(),
                user.getPseudo(),
                user.getName(),
                user.getFirstName(),
                user.getEmail(),
                user.getPassword(),
                user.getBirthday(),
                user.getBase64(),
                user.getLibrary(),
                user.getLiked()
        );
    }
}
